modules/backendscheduler/provider/compaction.go

```diff
@@ -361,6 +368,11 @@ func (p *CompactionProvider) newBlockSelector(tenantID string) (blockselector.Co
 	)
 
 	for _, job := range p.sched.ListJobs() {
+		// Clean up recent jobs cache when we find a job which has been persisted
+		p.recentJobsMtx.Lock()
+		delete(p.recentJobs, job.ID)
```

Comment:
Could there still be a gap where a block is no longer in ListJobs, and the new compacted status hasn't been polled yet?

Reply:
This is a good point.  It would be possible to configure the polling cycle and maintenance interval of the scheduler to be at odds with each-other.  I think we can solve this with a configuration validation to avoid this.  In practice, I don't expect this to come up and the default values are sane for the case you mention.

Reply:
Hmm, the config isn't plumbed in a way that makes this simple, since the store has the polling cycle.  I'll think about this a bit more.

Reply:
I've pushed a commit for the additional validation.